classes/Cart.php

```diff
@@ -1650,7 +1650,7 @@ public function deleteProduct(
         }
 
         $preservedGifts = $this->getProductsGifts($id_product, $id_product_attribute);
-        if ($preservedGifts[$id_product.'-'.$id_product_attribute] > 0) {
+        if (isset($preservedGifts[$id_product.'-'.$id_product_attribute]) && $preservedGifts[$id_product.'-'.$id_product_attribute] > 0) {
```

Comment:
Actually, a better would be a integer cast of the variable $id_product_attribute. The `isset()` here may prevent us to get the gifts in some case, although it exists.

```php
        $preservedGifts = $this->getProductsGifts($id_product, $id_product_attribute);
        if ($preservedGifts[$id_product.'-'. (int)$id_product_attribute] > 0) {
            return Db::getInstance()->execute(
                'UPDATE `'._DB_PREFIX_.'cart_product`
                SET `quantity` = '.(int)$preservedGifts[$id_product.'-'.(int)$id_product_attribute].'
                WHERE `id_cart` = '.(int)$this->id.'
                AND `id_product` = '.(int)$id_product.
                ($id_product_attribute != null ? ' AND `id_product_attribute` = '.(int)$id_product_attribute : '')
            );
        }
```

This caused by the default value `NULL` in the function parameters, although we expect a integer...
BTW, why the method `getProductsGifts` return an array while a integer would be enough.